core/client/src/main/java/alluxio/client/block/stream/LocalFilePacketWriter.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.block.stream;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+import alluxio.client.block.BlockWorkerClient;
+import alluxio.worker.block.io.LocalFileBlockWriter;
+
+import com.google.common.base.Preconditions;
+import io.netty.buffer.ByteBuf;
+
+import java.io.IOException;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * A local packet writer that simply writes packets to a local file.
+ */
+@NotThreadSafe
+public final class LocalFilePacketWriter implements PacketWriter {
+  private static final long PACKET_SIZE =
+      Configuration.getBytes(PropertyKey.USER_LOCAL_WRITER_PACKET_SIZE_BYTES);
+  private static final long FILE_BUFFER_BYTES =
+      Configuration.getBytes(PropertyKey.USER_FILE_BUFFER_BYTES);
+
+  /** The position to write the next byte at. */
+  private long mPos = 0;
+  /** The number of bytes reserved on the block worker to hold the block. */
+  private long mPosReserved = 0;
+  private final long mBlockId;
+  private final LocalFileBlockWriter mWriter;
+  private final BlockWorkerClient mBlockWorkerClient;
+  private boolean mClosed = false;
+
+  /**
+   * Creates an instance of {@link LocalFilePacketWriter}. This requires the block to be locked
+   * beforehand.
+   *
+   * @param blockWorkerClient the block worker client, not owned by this class
+   * @param blockId the block ID
+   * @throws IOException if it fails to create the packet writer
+   * @return the {@link LocalFilePacketWriter} created
+   */
+  public static LocalFilePacketWriter createLocalPacketWriter(BlockWorkerClient blockWorkerClient,
+      long blockId) throws IOException {
+    return new LocalFilePacketWriter(blockWorkerClient, blockId);
+  }
+
+  @Override
+  public long pos() {
+    return mPos;
+  }
+
+  @Override
+  public int packetSize() {
+    return (int) PACKET_SIZE;
```

Comment:
It is only used by the 'PacketOutStream' which uses this to determine the buffer size.

I added this because I might need to use a larger packet size for 'LocalFilePacketWriter' as its implementation is almost the same as what we had before this PR. So if we want to achieve the same performance, i need to increase the size here.